cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +799,313 @@ def _sqlite_dump(self):
+
+        # First, we'll get a list of all objects that are either
+        # (a) custom, no namespace
+        # (b) custom, with our namespace
+        # (c) not ours (standard or other package), but have fields with our namespace or no namespace
+        self.describes = {}  # Cache per-object describes for efficiency
+        for obj in self.global_describe["sobjects"]:
+            self.describes[obj["name"]] = getattr(self.sf, obj["name"]).describe()
+            if self._is_our_custom_api_name(obj["name"]) or self._has_our_custom_fields(
+                self.describes[obj["name"]]
+            ):
+                if self._is_object_mappable(obj):
+                    self.mapping_objects.append(obj["name"])
+
+        # Add any objects that are required by our own,
+        # meaning any object we are looking up to with a custom field,
+        # or any master-detail parent of any included object.
+        index = 0
+        while index < len(self.mapping_objects):
+            obj = self.mapping_objects[index]
+            for field in self.describes[obj]["fields"]:
+                if field["type"] == "reference":
+                    if field["relationshipOrder"] == 1 or self._is_any_custom_api_name(
+                        field["name"]
+                    ):
+                        self.mapping_objects.extend(
+                            filter(
+                                lambda o: o not in self.mapping_objects
+                                and self._is_object_mappable(self.describes[o]),
+                                field["referenceTo"],
+                            )
+                        )
+            index += 1
+
+    def _build_schema(self):
+        # Now, find all the fields we need to include.
+        # For custom objects, we include all custom fields. This includes custom objects
+        # that our package doesn't own.
+        # For standard objects, we include all custom fields, all required standard fields,
+        # and master-detail relationships. Required means createable and not nillable.
+        self.schema = {}
+        self.refs = {}
+        for obj in self.mapping_objects:
+            self.schema[obj] = {}
+
+            for field in self.describes[obj]["fields"]:
+                if any(
+                    [
+                        self._is_any_custom_api_name(field["name"]),
+                        self._is_core_field(field["name"]),
+                        self._is_required_field(field),
+                        self._is_lookup_to_included_object(field),
+                    ]
+                ):
+                    if self._is_field_mappable(obj, field):
+                        self.schema[obj][field["name"]] = field
+
+                        if field["type"] == "reference":
+                            for target in field["referenceTo"]:
+                                if (
+                                    obj in self.mapping_objects
```

Comment:
Is there a circumstance where `obj` wouldn't be in `self.mapping_objects`? I could be missing something, but it looks like `obj` would always be contained in `self.mapping_objects`

Reply:
Both halves of this `if` have been made redundant by checks elsewhere. Thanks!